homeassistant/components/ezviz/camera.py

```diff
@@ -0,0 +1,276 @@
+
+            cameras[device_serial] = {
+                "serial": device_serial,
+                "name": device["name"],
+                "deviceType": device["deviceType"],
+                "version": device["version"],
+                "status": device["status"],
+                "createTime": device["userDeviceCreateTime"],
+                "category": device["deviceCategory"],
+                "subCategory": device["deviceSubCategory"],
+                "customType": device["customType"],
+                "localIp": connections[device_serial]["localIp"],
+                "netIp": connections[device_serial]["netIp"],
+                "localRtspPort": local_rtsp_port,
+                "netType": connections[device_serial]["netType"],
+                "wanIp": connections[device_serial]["wanIp"],
+            }
+
+        # Add the cameras as devices in HASS
+        for camera_serial in cameras:
+            camera = cameras[camera_serial]
+            _LOGGER.debug("CAMERA: %s", camera)
+
+            camera_username = CONF_DEFAULT_CAMERA_USERNAME
+            camera_password = ""
+            camera_rtsp_stream = ""
+
+            if camera_serial in conf_cameras:
+                camera_username = conf_cameras[camera_serial]["username"]
+                camera_password = conf_cameras[camera_serial]["password"]
+                camera_rtsp_stream = "rtsp://{}:{}@{}:{}".format(
+                    camera_username,
+                    camera_password,
+                    camera["localIp"],
+                    camera["localRtspPort"],
+                )
+                _LOGGER.debug(
+                    "Camera %s source stream: %s", camera["serial"], camera_rtsp_stream
+                )
+
+            else:
+                _LOGGER.error(
+                    "I found a camera (%s) but it is not configured. Please configure it if you wish to see the appropriate stream. Conf cameras: %s",
+                    camera_serial,
+                    conf_cameras,
+                )
+
+            async_add_entities(
+                [
+                    EzvizCamera(
+                        hass,
+                        camera_username,
+                        camera_password,
+                        camera_rtsp_stream,
+                        camera["serial"],
+                        camera["name"],
+                        camera["deviceType"],
+                        camera["version"],
+                        camera["status"],
+                        camera["createTime"],
+                        camera["category"],
+                        camera["subCategory"],
+                        camera["customType"],
+                        camera["localIp"],
+                        camera["netIp"],
+                        camera["localRtspPort"],
+                        camera["wanIp"],
+                        camera["netType"],
+                    )
+                ]
+            )
+    except PyEzvizError as exp:
```

Comment:
Do not wrap this entire method in a `try/except`.  Protected explicit lines that may except and handle the exception appropriately.

Reply:
Ok I changed the code. May I ask why it is more appropriate ?

Reply:
It is good practice to catch specific exception types that wrap specific parts of the code.  This improves readability and maintainability, and allows you to take specific action, such as log a useful error message.  In this case, `PyEzvizError` can only be raised by LN51, LN54-55 why wrap the rest of the function?  It would be OK to wrap the try/except around those specific operations.